src/async/rest/signouts-tables.ts

```diff
@@ -0,0 +1,174 @@
+import { OptionalBoolean, OptionalNumber, OptionalString, makeOptional, makeOptionalProps, OptionalDateTime } from "util/OptionalTypeValidators";
+import * as t from "io-ts";
+import APIWrapper from "../../core/APIWrapper";
+import { HttpMethod } from "../../core/HttpMethod";
+import { optionifyProps } from "util/OptionifyObjectProps";
+import optionify from "util/optionify";
+import { isInteger } from "lodash";
+
+const pathGet = "/rest/signouts-today";
+const pathPost = "/rest/signout";
+const pathPostMulti = "/rest/signouts";
+const pathSignoutCrew = "/rest/signout-crew";
+const pathGetBoatTypes = "/rest/boat-types";
+const pathGetRatings = "/rest/ratings";
+const pathRunagroundCapsize = "/rest/runaground-capsize";
+const pathPersonByCardNumber = "/rest/person/by-card";
+
+const HRNames = {
+	signoutType: "Signout Type",
+	sailNumber: "Sail Number"
+}
+
+const StringValidate = new t.Type<string, string, unknown>(
+	'OptionalString',
+	(input: unknown): input is string => typeof input === 'string',
+	(input, context) => (typeof input === 'string' && input.trim().length>0 ? t.success(input) : t.failure(input, context, HRNames[context[1].key] + " cannot be blank")),
+	t.identity
+);
+
+const IntValidate = new t.Type<number, string, unknown>(
+	'number',
+	(input: unknown): input is number => typeof input === 'number',
+	(input, context) => {return (isInteger(Number(input)) ? t.success(Number(input)) : t.failure(input, context, HRNames[context[1].key] + " is an invalid number"))},
+	(a) => String(a)
+);
+
+export const personRatingValidator = t.type({
+	personId: t.number,
+	ratingId: t.number,
+	programId: t.number
+});
+
+export const crewPersonValidator = t.type({
+	personId: t.number,
+	nameFirst: t.string,
+	nameLast: t.string
+})
+
+export const signoutCrewValidator = t.type({
+	$$person: crewPersonValidator,
+	crewId: OptionalNumber,
+	signoutId: OptionalNumber,
+	cardNum: OptionalString,
+	personId: OptionalNumber,
+	startActive: OptionalDateTime,
+	endActive: OptionalDateTime
+});
+
+export const skipperValidator = t.type({
+	$$personRatings: t.array(personRatingValidator),
+	nameFirst: t.string,
+	nameLast: t.string,
+	personId: t.number
+});
+
+export const signoutValidator = t.type({
+	signoutId: t.number,
+	programId: t.number,
+	$$skipper: skipperValidator,
+	apAttendanceId: OptionalNumber,
+	jpAttendanceId: OptionalNumber,
+	boatId: t.number,
+	personId: OptionalNumber,
+	cardNum: OptionalString,
+	sailNumber: OptionalString,
+	hullNumber: OptionalString,
+	signoutDatetime: OptionalDateTime,
+	signinDatetime: OptionalDateTime,
+	testRatingId: OptionalNumber,
+	testResult: OptionalString,
+	signoutType: StringValidate,
```

Comment:
What is the function of StringValidate here?  Seems like this can just be t.string so it is required, and the type is string rather than Option<string>.  Is it to validate a null value on submit from the FE to the server and get a pretty error message?